src/SPC/doctor/item/LinuxMuslCheck.php

```diff
@@ -8,57 +8,122 @@
+                shell(true)->exec($prefix . $fix_path);
+            }
+            return true;
+        } catch (RuntimeException) {
+            return false;
+        }
     }
 
     /**
+     * @throws DownloaderException
      * @throws RuntimeException
-     * @noinspection PhpUnused
+     * @throws FileSystemException
      */
-    #[AsFixItem('fix-musl')]
-    public function fixMusl(array $distro): bool
+    public function installMuslWrapper(): void
     {
-        $rhel_install = 'wget https://musl.libc.org/releases/musl-1.2.4.tar.gz && tar -zxvf musl-1.2.4.tar.gz && \
-                         rm -f musl-1.2.4.tar.gz && cd musl-1.2.4 && 
-                         if [[ ! "$PATH" =~ (^|:)"/usr/local/musl/bin"(:|$) ]]; then echo "export PATH=/usr/local/musl/bin:$PATH" >> ~/.bash_profile
-                         fi && \
-                         ./configure --enable-wrapper=gcc && \
-                         make -j && make install && cd .. && rm -rf musl-1.2.4';
-        $install_cmd = match ($distro['dist']) {
-            'ubuntu', 'debian' => 'apt-get install musl musl-tools -y',
-            'alpine' => 'apk add musl musl-utils musl-dev',
-            'rhel' => $rhel_install,
-            'almalinux' => $rhel_install,
-            default => throw new RuntimeException('Current linux distro does not have an auto-install script for musl packages yet.'),
-        };
         $prefix = '';
         if (get_current_user() !== 'root') {
             $prefix = 'sudo ';
             logger()->warning('Current user is not root, using sudo for running command');
         }
-        try {
-            shell(true)->exec($prefix . $install_cmd);
-            return true;
-        } catch (RuntimeException) {
-            return false;
-        }
+        $musl_source = [
+            'type' => 'url',
+            'url' => 'https://musl.libc.org/releases/musl-1.2.4.tar.gz',
+        ];
+        Downloader::downloadSource('musl-1.2.4', $musl_source);
+        FileSystem::extractSource('musl-1.2.4', DOWNLOAD_PATH . '/musl-1.2.4.tar.gz');
+        $musl_gcc_install_cmd = 'cd ' . SOURCE_PATH . '/musl-1.2.4 && \
+                         ./configure --enable-wrapper=gcc && \
+                         make -j && make install';
+        shell(true)->exec($prefix . $musl_gcc_install_cmd);
+    }
+
+    /**
+     * @throws DownloaderException
+     * @throws RuntimeException
+     * @throws FileSystemException
+     */
+    public function installMuslCrossMake(): void
+    {
+        $arch = arch2gnu(php_uname('m'));
+        $musl_compile_source = [
+            'type' => 'url',
+            'url' => "https://musl.cc/{$arch}-linux-musl-cross.tgz",
+        ];
+        Downloader::downloadSource('musl-compile', $musl_compile_source);
+        FileSystem::extractSource('musl-compile', DOWNLOAD_PATH . "/{$arch}-linux-musl-cross.tgz");
+        shell(true)->exec('cp -rf ' . SOURCE_PATH . '/musl-compile/* /usr/local/musl');
```

Comment:
Is it possible to install cross-make musl in user directory?

Reply:
That's the default path for it. We could technically put it anywhere as long as we add the bin folder to the $PATH